src/psyclone/domain/common/transformations/alg_trans.py

```diff
@@ -0,0 +1,102 @@
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author R. W. Ford STFC Daresbury Lab
+
+'''Specialise generic PSyIR representing an algorithm layer to a
+PSyclone algorithm-layer-specific PSyIR which uses specialised classes.
+
+'''
+from psyclone.psyGen import Transformation
+from psyclone.psyir.nodes import Call, Routine, Container
+from psyclone.domain.common.transformations import InvokeTrans
+from psyclone.psyir.transformations import TransformationError
+
+
+class AlgTrans(Transformation):
+    '''Transform a generic PSyIR representation of the Algorithm layer to
+    a PSyclone version with specialised domain-specific nodes.
+
+    :param str invoke_name: the name used to specify an invoke \
+        call. This is an optional argument that defaults to 'invoke'.
+
+    '''
+    def __init__(self, invoke_name="invoke"):
+        self._invoke_trans = InvokeTrans()
+        self._invoke_name = invoke_name
+
+    def validate(self, node, options=None):
+        '''Validate the supplied PSyIR tree.
+
+        :param node: A PSyIR node that is the root of a PSyIR tree.
+        :type node: :py:class:`psyclone.psyir.node.Routine` or \
+            :py:class:`psyclone.psyir.node.Container`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :raises TransformationError: if the supplied node argument is \
+            not a Routine or a Container.
+        :raises TransformationError: if the supplied node argument has \
+            a parent.
+
+        '''
+        if not isinstance(node, (Routine, Container)):
+            raise TransformationError(
+                "Error in {0} transformation. The supplied call argument "
+                "should be a Routine or Container node but found '{1}'."
+                "".format(self.name, type(node).__name__))
+        if node.parent:
+            raise TransformationError(
+                "Error in {0} transformation. The supplied node should be the "
+                "root of a PSyIR tree but this node has a parent."
+                "".format(self.name))
+
+    def apply(self, psyir, options=None):
+        ''' apply transformation to the supplied psyir node '''
```

Comment:
Missing docstring parameters and proper capitalisation and punctuation.

Reply:
Oops.